src/engine/GroupBy.cpp

```diff
@@ -775,7 +775,7 @@ void GroupBy::findGroupedVariableImpl(
     std::variant<std::vector<ParentAndChildIndex>, OccurAsRoot>& substitutions,
     const Variable& groupedVariable) {
   if (auto value = hasType<sparqlExpression::VariableExpression>(expr)) {
-    auto variable = value.value()->value();
+    auto variable = value->value();
```

Comment:
If this is really the only usage of this function (I thought there used to be more, but it appears that the `hasAnyType`  is completely unused now),
then by all means, use `auto value = dynamic_cast<.....>`
And get rid of the rest altogether.

Reply:
Yes